platform/windows/display_server_windows.cpp

```diff
@@ -5627,6 +5640,12 @@ DisplayServer::WindowID DisplayServerWindows::_create_window(WindowMode p_mode,
 			wd_transient_parent->transient_children.insert(id);
 		}
 
+		wd.sharp_corners = p_flags & WINDOW_FLAG_SHARP_CORNERS_BIT;
+		{
+			DWM_WINDOW_CORNER_PREFERENCE value = wd.sharp_corners ? DWMWCP_DONOTROUND : DWMWCP_DEFAULT;
+			::DwmSetWindowAttribute(wd.hWnd, DWMWA_WINDOW_CORNER_PREFERENCE, &value, sizeof(value));
```

Comment:
This attribute is Windows 11 only, so it won't be defined in older SDKs, `display_server_windows.h` should have fallback defines for it and `DWMWCP_DONOTROUND`, `DWMWCP_DEFAULT` and `DWM_WINDOW_CORNER_PREFERENCE` type (like it's done with other similar types).

```
#ifndef DWMWA_WINDOW_CORNER_PREFERENCE
#define DWMWA_WINDOW_CORNER_PREFERENCE 33
#endif

... etc ...
```